arangod/Cache/BucketState.cpp

```diff
@@ -45,18 +45,17 @@ BucketState& BucketState::operator=(BucketState const& other) noexcept {
 }
 
 bool BucketState::isLocked() const noexcept {
-  return ((_state.load() & static_cast<std::uint32_t>(Flag::locked)) > 0);
+  return ((_state.load() & static_cast<FlagType>(Flag::locked)) > 0);
 }
 
 bool BucketState::lock(std::uint64_t maxTries) noexcept {
   uint64_t attempt = 0;
   while (attempt < maxTries) {
     // expect unlocked, but need to preserve migrating status
-    std::uint32_t current = _state.load(std::memory_order_relaxed);
-    std::uint32_t expected =
-        current & (~static_cast<std::uint32_t>(Flag::locked));
+    FlagType current = _state.load(std::memory_order_relaxed);
+    FlagType expected = current & (~static_cast<FlagType>(Flag::locked));
     if (current == expected) {
-      uint32_t desired = expected | static_cast<std::uint32_t>(Flag::locked);
+      uint32_t desired = expected | static_cast<FlagType>(Flag::locked);
```

Comment:
uint32_t is better because everything smaller implicitly casts to int

Reply:
What does that mean exactly? 
The stored `_state` value is supposed to fit into a uint16_t, because if we use a uint32_t for it, the struct would become to big.
So not exactly sure what you are suggesting exactly. Thanks for clarifying it.